opentreemap/treemap/udf.py

```diff
@@ -358,17 +369,24 @@ def _validate_and_update_choice(
                 {'datatype': [_("Can't change choices "
                                 "on a non-choice field")]})
 
-        if old_choice_value not in datatype['choices']:
+        choices = datatype['choices']
+        protected_choices = datatype.get('protected_choices', [])
+        all_choices = choices + protected_choices
```

Comment:
> So should we have 6 functions in total to accomplish this?
I think it is OK for `_validate_and_update_choice` to reach into the datatype dictionary when it needs to (it is a private helper method after all), so I don't think we need separate helper functions for `choices` vs. `protected_choices` that support collection UDFs. The goal is more to ensure that logic around which field goes first when you combine the lists is located in one place, so that we don't have things in different orders depending on the code path, and to keep things outside the UDFD class from reaching into the datatype dictionary unnecessarily.